lib/resty/redis.lua

```diff
@@ -138,10 +138,16 @@ function _M.connect(self, host, port_or_opts, opts)
     local ok, err
 
     if unix then
-        ok, err = sock:connect(host, port_or_opts)
         opts = port_or_opts
+        if opts and opts.server_name and not opts.pool then
+            opts.pool = opts.server_name .. ":" .. host
```

Comment:
Should we follow [http](https://github.com/ledgetech/lua-resty-http/pull/206/files#diff-654e4e7b65177f9d0e6eb1d4b1c259dbR133-R143) module, put `opt.backlog`, `opt.ssl` and `opt.ssl_verify` into pool name too?